src/passes/TypeMerging.cpp

```diff
@@ -242,27 +242,34 @@ void TypeMerging::run(Module* module_) {
   // Merging can unlock more sibling merging opportunities because two identical
   // types cannot be merged until their respective identical parents have been
   // merged in a previous step, making them siblings.
-  //
-  // If we merge siblings, we also need to refinalize because the LUB of merged
-  // siblings is the merged type rather than their common supertype after the
-  // merge.
-  bool refinalize = false;
   merge(Supertypes);
   for (int i = 0; i < MAX_ITERATIONS; ++i) {
-    if (!merge(Siblings)) {
-      break;
-    }
-    refinalize = true;
+    merge(Siblings);
   }
 
   applyMerges();
 
-  if (refinalize) {
-    ReFinalize().run(getPassRunner(), module);
-  }
+  // If we merge siblings, we also need to refinalize because the LUB of merged
```

Comment:
Let's add a comment about why merging parents can require refinalization as well. (That will make the "also" make more sense here, too.)

Reply:
Sorry, I don't follow. The existing comment explained refinalization after `merge(Siblings)`, and the new comment, for `merge(Supertypes)`. Are those not the two cases? I'm not sure what else is unexplained.